kusto/ingest/internal/properties/properties.go

```diff
@@ -68,101 +72,140 @@ func (d DataFormat) CamelCase() string {
-	Raw DataFormat = 7
+	Raw DataFormat = 9
 	// SCSV is a file containing semicolon ";" separated values.
-	SCSV DataFormat = 8
+	SCSV DataFormat = 10
 	// SOHSV is a file containing SOH-separated values(ASCII codepont 1).
-	SOHSV DataFormat = 9
-	// TSV is a file containing table seperated values ("\t").
-	TSV DataFormat = 10
+	SOHSV DataFormat = 11
+	// SINGLEJSON indicates the source containg a single Javascript Object Notation encoded record, newlines are treated as whitespace
+	SINGLEJSON DataFormat = 12
+	// SSTREAM indicats the source is encoded as a Microsoft Cosmos Structured Streams format
+	SSTREAM DataFormat = 13
+	// TSV is a file containing tab seperated values ("\t").
+	TSV DataFormat = 14
+	// TSVE is a file containing escaped-tab seperated values ("\t").
+	TSVE DataFormat = 15
 	// TXT is a text file with lines deliminated by "\n".
-	TXT DataFormat = 11
+	TXT DataFormat = 16
+	// WCLOGFILE indicates the source is encoded using W3C Extended Log File format
+	WCLOGFILE DataFormat = 17
 )
 
+type dataFormatDescriptior struct {
+	CamelCaseName    string
+	JSONName         string
+	Extension        string
+	MappingRequired  bool
+	ValidMappingKind bool
+}
+
+var dataFormatDescriptionVec = []dataFormatDescriptior{
+	{"", "", "", false, false},
+	{"Avro", "avro", ".avro", true, true}, // Ke Avro before ApacheAvro so that fileformat detection would prefer it
+	{"ApacheAvro", "avro", ".avro", true, false},
+	{"Csv", "csv", ".csv", false, true},
+	{"Json", "json", ".json", true, true}, // Keep Json before Single or Multi Json, so that fileformat detection would prefer it
+	{"MultiJson", "json", ".json", true, false},
+	{"Orc", "orc", ".orc", false, true},
+	{"Parquet", "parquet", ".parquet", false, true},
+	{"Psv", "psv", ".psv", false, false},
+	{"Raw", "raw", ".raw", false, false},
+	{"Scsv", "scsv", ".scsv", false, false},
+	{"Sohsv", "sohsv", ".sohsv", false, false},
+	{"SingleJson", "json", ".json", true, false},
+	{"SStream", "sstream", ".sstream", false, false},
+	{"Tsv", "tsv", ".tsv", false, false},
+	{"Tsve", "tsve", ".tsve", false, false},
+	{"Txt", "txt", ".txt", false, false},
+	{"EcLogFile", "eclogfile", ".eclogfile", false, false},
+}
+
 // dfToExt takes a DataFormat and returns the file extension that it would use.
-func dfToExt(et DataFormat) (string, error) {
-	switch et {
-	case CSV:
-		return "csv", nil
-	case JSON:
-		return "json", nil
-	case AVRO:
-		return "avro", nil
-	case Parquet:
-		return "parquet", nil
-	case ORC:
-		return "orc", nil
-	case PSV:
-		return "psv", nil
-	case Raw:
-		return "raw", nil
-	case SCSV:
-		return "scsv", nil
-	case SOHSV:
-		return "sohsv", nil
-	case TSV:
-		return "tsv", nil
-	case TXT:
-		return "txt", nil
-	default:
-		return "", fmt.Errorf("EncodingType(%d) was no one we understand", et)
+func dfGetJSONName(et DataFormat) (string, error) {
+	if et > 0 && int(et) < len(dataFormatDescriptionVec) {
+		return dataFormatDescriptionVec[et].JSONName, nil
```

Comment:
dataFormatDescriptionVec 

That is more like java naming conventions, which is anti-goish.

We don't include types in names, so it is verboten to do:  myMapVar or in this case Vec.

dfDesc or something should be enough.

Reply:
dataFormatDescriptions